restcomm/restcomm.dao/src/main/java/org/mobicents/servlet/restcomm/dao/mybatis/MybatisGeolocationDao.java

```diff
@@ -0,0 +1,213 @@
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeDateTime;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeSid;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeUri;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readGeolocationType;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readLong;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.joda.time.DateTime;
+import org.mobicents.servlet.restcomm.dao.GeolocationDao;
+import org.mobicents.servlet.restcomm.entities.Geolocation;
+import org.mobicents.servlet.restcomm.entities.Sid;
+
+/**
+ * @author fernando.mendioroz@telestax.com (Fernando Mendioroz)
+ *
+ */
+public class MybatisGeolocationDao implements GeolocationDao {
+
+    private static final String namespace = "org.mobicents.servlet.sip.restcomm.dao.GeolocationDao.";
+    private final SqlSessionFactory sessions;
+
+    public MybatisGeolocationDao(final SqlSessionFactory sessions) {
+        super();
+        this.sessions = sessions;
+    }
+
+    @Override
+    public void addGeolocation(Geolocation gl) {
+        final SqlSession session = sessions.openSession();
+        try {
+            session.insert(namespace + "addGeolocation", toMap(gl));
+            session.commit();
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public Geolocation getGeolocation(Sid sid) {
+        return getGeolocation(namespace + "getGeolocation", sid.toString());
+    }
+
+    private Geolocation getGeolocation(final String selector, final String parameter) {
+        final SqlSession session = sessions.openSession();
+        try {
+            final Map<String, Object> result = session.selectOne(selector, parameter);
+            if (result != null) {
+                return toGeolocation(result);
+            } else {
+                return null;
+            }
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public List<Geolocation> getGeolocations(Sid accountSid) {
+        final SqlSession session = sessions.openSession();
+        try {
+            final List<Map<String, Object>> results = session.selectList(namespace + "getGeolocations", accountSid.toString());
+            final List<Geolocation> geolocations = new ArrayList<Geolocation>();
+            if (results != null && !results.isEmpty()) {
+                for (final Map<String, Object> result : results) {
+                    geolocations.add(toGeolocation(result));
+                }
+            }
+            return geolocations;
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public void removeGeolocation(Sid sid) {
+        removeGeolocations(namespace + "removeGeolocation", sid);
```

Comment:
Yes... but as told in previous comment, for now I would rather sticking to the same design pattern used throughout these classes.